### PR TITLE
Adding ListDefinitions endpoint

### PR DESCRIPTION
**Pull request by andresuribe87**
See title. Pretty straighforward change. 

Note that some formatting seems to be out of sync when running `mage spec`. This change also fixes that.

### CODE REVIEW

File: pkg/service/presentation/model/model.go [hunk 1]
```diff
@@ -78,6 +78,10 @@ type ListSubmissionResponse struct {
 	Submissions []Submission `json:"submissions"`
 }
 
+type ListDefinitionsResponse struct {
+	Definitions []*exchange.PresentationDefinition `json:"definitions"`
```
Comment (decentralgabe) [inline pkg/service/presentation/model/model.go:82]: is there a need for this to be a pointer?

**Review by thereisnogabe — APPROVED** [review]

File: pkg/service/presentation/model/model.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/service/presentation/model/model.go:82]: No need, but seemed unnecessary to copy the full struct.

File: pkg/server/server_presentation_test.go [hunk 2]
```diff
@@ -105,6 +101,39 @@ func TestPresentationAPI(t *testing.T) {
...
+
+		req := httptest.NewRequest(http.MethodGet, "https://ssi-service.com/v1/presentations/definitions", nil)
+		w := httptest.NewRecorder()
+
+		assert.NoError(tt, pRouter.ListDefinitions(newRequestContext(), w, req))
+
+		var resp router.ListDefinitionsResponse
+		assert.NoError(tt, json.NewDecoder(w.Body).Decode(&resp))
+		assert.Len(tt, resp.Definitions, 2)
+		assert.ElementsMatch(tt, resp.Definitions, []*exchange.PresentationDefinition{
```
[the remark this answers is not in the document]
Comment (andresuribe87) [reply pkg/server/server_presentation_test.go:131]: 🍾

File: pkg/server/router/presentation.go [hunk 3]
```diff
@@ -258,18 +296,20 @@ func (pr PresentationRouter) CreateSubmission(ctx context.Context, w http.Respon
 
 type GetSubmissionResponse struct {
 	*model.Submission
+	// TODO: Actually add the credentials that were sent.
```
[the remark this answers is not in the document]
Comment (andresuribe87) [reply pkg/server/router/presentation.go]: Added one in linear only. It should be quick.

File: pkg/server/router/credential.go [hunk 4]
```diff
@@ -303,15 +308,16 @@ type VerifyCredentialResponse struct {
...
-// @Tags         CredentialAPI
-// @Accept       json
-// @Produce      json
-// @Param        request  body      VerifyCredentialRequest  true  "request body"
-// @Success      200  {object}  VerifyCredentialResponse
-// @Failure      400  {string}  string  "Bad request"
-// @Router       /v1/credentials/verification [put]
+//
+//	@Summary		Verify Credential
+//	@Description	Verify a given credential by its id
```
[the remark this answers is not in the document]
Comment (andresuribe87) [reply pkg/server/router/credential.go:313]: Absolutely! Going to leave for a separate PR that audits all documentation.

**Pull request merged**